vmm/src/vmm_config/drive.rs

```diff
@@ -158,72 +157,84 @@ impl BlockDeviceConfigs {
     }
 
     fn create(&mut self, block_device_config: BlockDeviceConfig) -> Result<()> {
-        // check if the path exists
+        // If no path exists, then an error is returned.
         if !block_device_config.path_on_host.exists() {
-            return Err(DriveError::InvalidBlockDevicePath);
+            Err(DriveError::InvalidBlockDevicePath)
         }
-
-        if self
+        // If the path already exists, then an error is returned.
+        else if self
```

Comment:
no need to have it as an else-if.

Reply:
In this case, what I was trying to do is to replace a chain of if ... {return ...} if ... {return ... } etc. statemenents, with a single if ... { ... } else if ... { ... } else if ... { ... } etc. expression. In light of this the else if is necessary. This is mostly because in this way it's syntactically clear that the conditioning is exhaustive (since an if-else if -else if ... else expression must always return a value). I prefer this for the same reason I would prefer ```return match x { A => ... ; B => ... } ``` to ```if x == A { return ... } if x == B {return ... }```

Reply:
Yeah, this is probably just the developer preference. You can keep the change if you want to.